applications/FluidDynamicsApplication/custom_utilities/fluid_auxiliary_utilities.cpp

```diff
@@ -0,0 +1,222 @@
+    const auto& r_communicator = rModelPart.GetCommunicator();
+    if (r_communicator.LocalMesh().NumberOfNodes() !=0) {
+        KRATOS_ERROR_IF_NOT(r_communicator.LocalMesh().NodesBegin()->SolutionStepsDataHas(DISTANCE)) << "Nodal solution step data has no \'DISTANCE\' variable. Negative volume cannot be computed" << std::endl;
+    }
+
+    double fluid_volume = 0.0;
+    if (r_communicator.LocalMesh().NumberOfElements() != 0) {
+        // Set the modified shape functions fatory
+        const auto& r_geom_begin = r_communicator.LocalMesh().ElementsBegin()->GetGeometry();
+        auto mod_sh_func_factory = GetStandardModifiedShapeFunctionsFactory(r_geom_begin);
+
+        // Calculate the negative volume
+        Vector nodal_distances(r_geom_begin.PointsNumber());
+        fluid_volume = block_for_each<SumReduction<double>>(rModelPart.GetCommunicator().LocalMesh().Elements(), nodal_distances, [&mod_sh_func_factory](Element& rElement, Vector& rNodalDistancesTLS){
+            // Set the distances vector to check if the element is split
+            const auto& r_geom = rElement.GetGeometry();
+            const std::size_t n_nodes = r_geom.PointsNumber();
+            for (std::size_t i_node = 0; i_node < n_nodes; ++i_node) {
+                rNodalDistancesTLS[i_node] = r_geom[i_node].FastGetSolutionStepValue(DISTANCE);
+            }
+            // Split element check
+            double elem_volume = 0.0;
+            if (IsSplit(rNodalDistancesTLS)) {
+                // Compute negative volume fraction with the modified shape functions
+                auto p_mod_sh_func = mod_sh_func_factory(rElement.pGetGeometry(), rNodalDistancesTLS);
+                elem_volume = p_mod_sh_func->ComputeNegativeSideDomainSize();
+            } else if (IsNegative(rNodalDistancesTLS)) {
+                // If the element is negative, compute the volume from geometry
+                elem_volume = r_geom.DomainSize();
+            }
+
+            // Return the value to be reduced
+            return elem_volume;
+        });
+    }
+
+    // Synchronize among processors
+    r_communicator.GetDataCommunicator().SumAll(fluid_volume);
+
+    return fluid_volume;
+}
+
+double FluidAuxiliaryUtilities::CalculateFlowRate(const ModelPart& rModelPart)
+{
+    // Check that there are conditions and distance variable in the nodal database
+    KRATOS_ERROR_IF(rModelPart.GetCommunicator().GlobalNumberOfConditions() == 0) << "There are no conditions in the provided model part. Flow rate cannot be computed." << std::endl;
+    const auto& r_communicator = rModelPart.GetCommunicator();
+    if (r_communicator.LocalMesh().NumberOfNodes() !=0) {
+        KRATOS_ERROR_IF_NOT(r_communicator.LocalMesh().NodesBegin()->SolutionStepsDataHas(VELOCITY)) << "Nodal solution step data has no \'VELOCITY\' variable. Flow rate cannot be computed" << std::endl;
```

Comment:
All incompressible formulations in `FluidDynamicsApplication` assume `VELOCITY` as variable. Compressible ones are written in conservative variables so this would require a different method to calculate the flow rate.

Having said so, I think that making it optional is not a real advantage in this case, specially considering that it would require to always use kwargs in the Python export.